Database/MongoDB/Internal/Protocol.hs

```diff
@@ -466,17 +736,38 @@ replyOpcode = 1
 getReply :: Get (ResponseTo, Reply)
 getReply = do
     (opcode, responseTo) <- getHeader
-    unless (opcode == replyOpcode) $ fail $ "expected reply opcode (1) but got " ++ show opcode
-    rResponseFlags <-  rFlags <$> getInt32
-    rCursorId <- getInt64
-    rStartingFrom <- getInt32
-    numDocs <- fromIntegral <$> getInt32
-    rDocuments <- replicateM numDocs getDocument
-    return (responseTo, Reply{..})
+    if opcode == 2013
+      then do
+            -- Notes:
+            -- Checksum bits that are set by the server don't seem to be supported by official drivers.
+            -- See: https://github.com/mongodb/mongo-python-driver/blob/master/pymongo/message.py#L1423
+            flagBits <-  rFlagsOpMsg <$> getInt32
+            _ ← getInt8
```

Comment:
It looks like you use a mix of UTF 8 chars and regular ascii chars. Could you please use only ascii characters?

On a personal note. What editor do you use to deal with them comfortably?

Reply:
Yes, I usually use haskell's unicode support (enabled with  `{-# LANGUAGE UnicodeSyntax #-}`)  and I edit in emacs. Works great. But you're right I forgot to switch back to ascii. I'll push an update once I figured out how to do this efficiently.